src/main/java/gregtech/api/metatileentity/implementations/MTEMultiBlockBase.java

```diff
@@ -2072,18 +2072,24 @@ public void getWailaNBTData(EntityPlayerMP player, TileEntity tile, NBTTagCompou
         tag.setBoolean("incompleteStructure", (getBaseMetaTileEntity().getErrorDisplayID() & 64) != 0);
 
         if (mOutputItems != null) {
-            tag.setInteger("outputItemLength", mOutputItems.length);
-            for (int i = 0; i < mOutputItems.length; i++) {
-                tag.setString("outputItem" + i, mOutputItems[i].getDisplayName());
-                tag.setInteger("outputItemCount" + i, mOutputItems[i].stackSize);
+            int index = 0;
+            for (ItemStack stack : mOutputItems) {
+                if (stack == null) continue;
+                tag.setString("outputItem" + index, mOutputItems[index].getDisplayName());
```

Comment:
Should probably be (along with the other three array accesses):
```suggestion
                tag.setString("outputItem" + index, stack.getDisplayName());
```